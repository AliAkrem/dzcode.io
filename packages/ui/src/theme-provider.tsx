import { CssBaseline, Direction, ThemeProvider as MuiThemeProvider } from "@mui/material";

import { FC } from "react";
import { PluginsProvider } from "./plugins-provider";
import { darkTheme } from "./dark-theme";
import { lightTheme } from "./light-theme";

interface Props {
  mode?: "dark" | "light";
  direction: Direction;
}

const ThemeProvider: FC<Props> = ({ children, mode, direction }) => {
  return (
    <MuiThemeProvider theme={mode === "dark" ? darkTheme(direction) : lightTheme(direction)}>
      <PluginsProvider direction={direction}>
        <CssBaseline />
        {children}
      </PluginsProvider>
    </MuiThemeProvider>
  );
};

export default ThemeProvider;
