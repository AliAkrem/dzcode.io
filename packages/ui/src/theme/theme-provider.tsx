import type { Direction } from "@mui/material";
import CssBaseline from "@mui/material/CssBaseline";
import MuiThemeProvider from "@mui/material/styles/ThemeProvider";
import { FC } from "react";

import { darkTheme } from "./dark-theme";
import { lightTheme } from "./light-theme";
import { PluginsProvider } from "./plugins-provider";

interface Props {
  mode?: "dark" | "light";
  direction?: Direction;
}

export const ThemeProvider: FC<Props> = ({ children, mode, direction = "ltr" }) => {
  return (
    <PluginsProvider direction={direction}>
      <MuiThemeProvider theme={mode === "dark" ? darkTheme(direction) : lightTheme(direction)}>
        <CssBaseline />
        {children}
      </MuiThemeProvider>
    </PluginsProvider>
  );
};
