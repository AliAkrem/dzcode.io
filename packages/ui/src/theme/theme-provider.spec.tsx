import { Button } from "@mui/material";
import { render } from "@testing-library/react";

import { ThemeProvider } from "./theme-provider";

it("should render dzcode light theme", () => {
  const { container } = render(
    <ThemeProvider mode="light">
      <Button variant="contained">Button</Button>
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});

it("should render dzcode dark theme", () => {
  const { container } = render(
    <ThemeProvider mode="dark">
      <Button variant="contained">Button</Button>
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});

it("should render inverted theme", () => {
  const { container } = render(
    <ThemeProvider direction="rtl">
      <Button variant="contained">اضغظ هنا</Button>
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});
