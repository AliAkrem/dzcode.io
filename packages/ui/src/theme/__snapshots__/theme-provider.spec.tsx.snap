// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render dzcode dark theme 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  muiltr-7ifnzh-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`should render dzcode light theme 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  muiltr-q9vf94-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`should render inverted theme 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  muirtl-q9vf94-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    اضغظ هنا
  </button>
</div>
`;
